app/database/models/server/thread.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Query, Relation} from '@nozbe/watermelondb';
+import {children, field, immutableRelation} from '@nozbe/watermelondb/decorators';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+
+import {MM_TABLES} from '@constants/database';
+
+import type PostModel from '@typings/database/models/servers/post';
+import type ThreadParticipantModel from '@typings/database/models/servers/thread_participant';
+
+const {POST, THREAD, THREAD_PARTICIPANT} = MM_TABLES.SERVER;
+
+/**
+ * The Thread model contains thread information of a post.
+ */
+export default class ThreadModel extends Model {
+    /** table (name) : Thread */
+    static table = THREAD;
+
+    /** associations : Describes every relationship to this table. */
+    static associations: Associations = {
+
+        /** A THREAD is associated to one POST (relationship is 1:1) */
+        [POST]: {type: 'belongs_to', key: 'id'},
```

Comment:
I'm doubtful about this one.. but seems to be working.. not sure is needed but 0/5 if it does not cause any problems

Reply:
After we fetch the threads for `global threads screen`, actual thread/post info is fetched from the post. I am observing it and getting the post data.

Reply:
can you share that code so I can take a look.. my doubt is not about the need for the association but more on how it was done as both tables seem to have `belongs_to`

Reply:
https://github.com/mattermost/mattermost-mobile/blob/dd70a51b12a3c4aab57a1a3db33fef79be1b9af2/app/screens/global_threads/threads_list/thread/index.ts